controllers/machineset_controller.go

```diff
@@ -443,7 +443,13 @@ func shouldExcludeMachine(machineSet *clusterv1.MachineSet, machine *clusterv1.M
 		logger.V(4).Info("Machine is not controlled by machineset", "machine", machine.Name)
 		return true
 	}
-	return !machine.ObjectMeta.DeletionTimestamp.IsZero()
+
+	// even machines in deleting status should be included when calculate replicas
+	if !machine.ObjectMeta.DeletionTimestamp.IsZero() {
+		logger.V(4).Info("Include Machine in deleting status", "machine", machine.Name)
+	}
+
+	return false
```

Comment:
It looks like this change could affect the logic for remediation above on lines 232-245, but I'm not entirely sure.